From efa67bb971458f14bc201a423a2fcd4983a6228d Mon Sep 17 00:00:00 2001
From: Alexandre Belloni <alexandre.belloni@free-electrons.com>
Date: Wed, 10 Aug 2016 12:00:20 +0200
Subject: [PATCH] Fix building without OpenGL support

Commit 2a7cee47e5e8 ("Avoid artifacts when hiding or closing a
QOpenGLWidget or QQuickWidget child") moved the
qt_dummy_platformTextureList definition outside of #ifndef
QT_NO_OPENGL/#endif

This results in the following error when building without OpenGL:
In file included from ../../include/QtCore/qglobalstatic.h:1:0,
                 from ../../include/QtCore/../../../git/src/corelib/global/qglobal.h:1111,
                 from ../../include/QtCore/qglobal.h:1,
                 from mkspecs/linux-oe-g++/qplatformdefs.h:47,
                 from src/widgets/kernel/qwidgetbackingstore.cpp:35:
src/widgets/kernel/qwidgetbackingstore.cpp: In function '{anonymous}::Q_QGS_qt_dummy_platformTextureList::Type* {anonymous}::Q_QGS_qt_dummy_platformTextureList::innerFunction()':
../../include/QtCore/../../../git/src/corelib/global/qglobalstatic.h:81:18: error: field 'value' has incomplete type '{anonymous}::Q_QGS_qt_dummy_platformTextureList::Type {aka QPlatformTextureList}'
             Type value;                                         \
                  ^
../../include/QtCore/../../../git/src/corelib/global/qglobalstatic.h:146:9: note: in expansion of macro 'Q_GLOBAL_STATIC_INTERNAL'
         Q_GLOBAL_STATIC_INTERNAL(ARGS)                                      \
         ^
../../include/QtCore/../../../git/src/corelib/global/qglobalstatic.h:153:5: note: in expansion of macro 'Q_GLOBAL_STATIC_WITH_ARGS'
     Q_GLOBAL_STATIC_WITH_ARGS(TYPE, NAME, ())
     ^
src/widgets/kernel/qwidgetbackingstore.cpp:64:1: note: in expansion of macro 'Q_GLOBAL_STATIC'
 Q_GLOBAL_STATIC(QPlatformTextureList, qt_dummy_platformTextureList)
 ^

Signed-off-by: Alexandre Belloni <alexandre.belloni@free-electrons.com>
---
 src/widgets/kernel/qwidgetbackingstore.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/widgets/kernel/qwidgetbackingstore.cpp b/src/widgets/kernel/qwidgetbackingstore.cpp
index 3f62e79..557e710 100644
--- a/src/widgets/kernel/qwidgetbackingstore.cpp
+++ b/src/widgets/kernel/qwidgetbackingstore.cpp
@@ -61,7 +61,9 @@ QT_BEGIN_NAMESPACE
 
 extern QRegion qt_dirtyRegion(QWidget *);
 
+#ifndef QT_NO_OPENGL
 Q_GLOBAL_STATIC(QPlatformTextureList, qt_dummy_platformTextureList)
+#endif
 
 /**
  * Flushes the contents of the \a backingStore into the screen area of \a widget.
-- 
1.9.1

